source/tilecameragenerator.ts

```diff
@@ -0,0 +1,491 @@
+ * let offset: [number, number];
+ *
+ * iteration variant 1:
+ * for (let i = 0; i < tileCameraGenerator.numberOfTiles(); ++i){
+ *      tileCameraGenerator.tile = i; // property
+ *      tileCameraGenerator.update();
+ *      offset = tileCameraGenerator.offset;
+ *      "render camera"
+ * }
+ *
+ * iteration variant 2:
+ * while(tileCameraGenerator.nextTile()){
+ *      offset = tileCameraGenerator.offset;
+ *      "render"
+ * }
+ * // reset generator
+ * tileCameraGenerator.resetTileRendering();
+ *
+ * NOTE: Use updateCameraProperties if the source camera is altered.
+ */
+
+
+export class TileCameraGenerator {
+
+    /** @see {@link eye} */
+    protected _camera: Camera | undefined;
+
+    /** @see {@link sourceCamera} */
+    protected _sourceCamera: Camera | undefined;
+
+    /** @see {@link tile} */
+    protected _tile = -1;
+
+    /** @see {@link sourceViewPort} */
+    protected _sourceViewPort: [number, number] | undefined;
+
+    /** @see {@link tileSize} */
+    protected _tileSize: [number, number] | undefined;
+
+    /** @see {@link padding} */
+    protected _padding: vec4 = vec4.fromValues(0, 0, 0, 0);
+
+    /** @see {@link algorithm} */
+    protected _algorithm: TileCameraGenerator.IterationAlgorithm = TileCameraGenerator.IterationAlgorithm.ScanLine;
+
+    protected _valid: boolean;
+    protected _currentOffset: [number, number] = [0, 0];
+    protected _iterationAlgorithmIndices: [number, number][] = [];
+
+    /**
+     * Converts the tile index from the selected Algorithm to table indices.
+     * @returns - The converted tile index.
+     */
+    protected tileIndexToTableIndices(): [number, number] {
+        if (this.algorithm === TileCameraGenerator.IterationAlgorithm.ScanLine) {
+            // Calculate column (0) and row (1) index of the current tile.
+            const index = this.tile;
+            const tableIndices: [number, number] = [0, 0];
+            tableIndices[0] = index % this.numberOfXTiles();
+            tableIndices[1] = Math.floor((index - tableIndices[0]) / this.numberOfXTiles());
+            return tableIndices;
+        } else if (this.algorithm === TileCameraGenerator.IterationAlgorithm.HilbertCurve) {
+            if (this._iterationAlgorithmIndices.length === 0) {
+                this.fillIterationIndicesWithHilbert();
+            }
+            return this._iterationAlgorithmIndices[this.tile];
+        } else if (this.algorithm === TileCameraGenerator.IterationAlgorithm.ZCurve) {
+            if (this._iterationAlgorithmIndices.length === 0) {
+                this.fillIterationIndicesWithZCurve();
+            }
+            return this._iterationAlgorithmIndices[this.tile];
+        } else {
+            return [0, 0];
+        }
+    }
+
+    /**
+     * Fills the iterationAlgorithmIndices with the table indices
+     * from the ZCurve-Iteration-Algorithm.
+     */
+    protected fillIterationIndicesWithZCurve(): void {
+        this._iterationAlgorithmIndices = [];
```

Comment:
I suggest to don't use the `push` interface of an array but initialize the length up-front and assign the values instead of pushing them at the end of the array.
If the final number of entries is not known up-front, compute the upper bound and resize the array down afterwards.